### PR TITLE
Fix deprecations on PHP 8.2

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | On the road to #44282
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Code is now 100% deprecation-free on PHP 8.2.
Changes on VarExporter are required to work around https://github.com/php/php-src/issues/8152

File: src/Symfony/Component/VarExporter/Tests/VarExporterTest.php [hunk 1]
```diff
@@ -127,9 +129,15 @@ public function provideExport()
...
-        $value = new \ArrayObject();
+        yield ['datetime', [
+            \DateTime::createFromFormat('U', 0),
+            \DateTimeImmutable::createFromFormat('U', 0),
+            new \DateTimeZone('Europe/Paris'),
+            new \DateInterval('P7D'),
+            new \DatePeriod('R4/2012-07-01T00:00:00Z/P7D'),
+        ]];
+
+        $value = \PHP_VERSION_ID >= 70406 ? new ArrayObject() : new \ArrayObject();
```
Comment (derrabus) [inline src/Symfony/Component/VarExporter/Tests/VarExporterTest.php:140]: I'm curious, what's the reason for this version check?

Comment (nicolas-grekas) [reply src/Symfony/Component/VarExporter/Tests/VarExporterTest.php:140]: PHP 8.2 needs the attribute for this test but I also want to keep a test that creates a native ArrayObject.

Comment (derrabus) [reply src/Symfony/Component/VarExporter/Tests/VarExporterTest.php:140]: Oh, I see. That version number 7.4.6 seemed so oddly specific that I though I had missed something in PHP's ChangeLog. 😅

**Pull request merged**